connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -401,19 +403,24 @@ public String toString() {
     }
 
     private void convertMessages(ConsumerRecords<byte[], byte[]> msgs) {
+        origOffsets.clear();
         for (ConsumerRecord<byte[], byte[]> msg : msgs) {
             log.trace("Consuming message with key {}, value {}", msg.key(), msg.value());
             SchemaAndValue keyAndSchema = keyConverter.toConnectData(msg.topic(), msg.key());
             SchemaAndValue valueAndSchema = valueConverter.toConnectData(msg.topic(), msg.value());
-            SinkRecord record = new SinkRecord(msg.topic(), msg.partition(),
+            SinkRecord origRecord = new SinkRecord(msg.topic(), msg.partition(),
                     keyAndSchema.schema(), keyAndSchema.value(),
                     valueAndSchema.schema(), valueAndSchema.value(),
                     msg.offset(),
                     ConnectUtils.checkAndConvertTimestamp(msg.timestamp()),
                     msg.timestampType());
-            record = transformationChain.apply(record);
-            if (record != null) {
-                messageBatch.add(record);
+            SinkRecord transRecord = transformationChain.apply(origRecord);
+            if (transRecord != null) {
```

Comment:
I considered this as an optimization. In any case, it was already there since the initial commit.

I'm inclined to commit all the offsets.